brian2/devices/device.py

```diff
@@ -315,7 +336,9 @@ def reinit(self):
         Reinitialize the device. For standalone devices, clears all the internal
         state of the device.
         '''
-        pass
+        # Dummy call to set_device with the currently set device -- will trigger
+        # a new activate, important if we are reinitializing the current device
+        set_device(get_device())
```

Comment:
:-) Reading my own comment, it seems that `get_device().activate()` would make more sense here, actually.